bibformat/format_elements/bfe_INSPIRE_enhanced_marcxml.py

```diff
@@ -0,0 +1,200 @@
+    citation_element = {}
+    for code, value in subfields:
+        if code == 'a':
+            citation_element['doi_string'] = value
+        elif code == 't':
+            citation_element['title'] = value
+        elif code == 'i':
+            citation_element['isbn'] = value
+        elif code == 'r':
+            citation_element['report_num'] = value
+        elif code == 'y':
+            citation_element['year'] = value
+        elif code == 's':
+            try:
+                journal, volume, page = value.split(',')
+                citation_element['journal_title'] = journal
+                citation_element['volume'] = volume
+                citation_element['page'] = page
+            except ValueError:
+                pass
+    if 'journal_title' in citation_element:
+        # HACK: to work around clash between find_journal() and find_book()
+        citation_element['title'] = citation_element['journal_title']
+    return citation_element
+
+
+def get_matched_id(subfields):
+    citation_element = reference2citation_element(subfields)
+    if 'doi_string' in citation_element:
+        recids = find_doi(citation_element)
+        if len(recids) == 1:
+            return recids.pop()
+    if 'journal_title' in citation_element and 'year' in citation_element:
+        recids = find_journal(citation_element)
+        if len(recids) == 1:
+            return recids.pop()
+    if 'report_num' in citation_element:
+        recids = find_reportnumber(citation_element)
+        if len(recids) == 1:
+            return recids.pop()
+    if 'isbn' in citation_element:
+        recids = find_isbn(citation_element)
+        if len(recids) == 1:
+            return recids.pop()
+    #if 'title' in citation_element:
+        #recids = find_book(citation_element)
+        #if len(recids) == 1:
+            #return recids.pop()
+    return None
+
+
+def format_element(bfo, oai=0):
+    """Produce MARCXML with enhanced fields.
+
+    Adds 100/700 $x with Record ID of linked HepName,
+                 $y with True/False if the signature is claimed
+                 $z with Record ID of institution
+         371/110 $z with Record ID of institution
+         502     $z with Record ID of institution
+         999C5   $0 with on the fly discovered Record IDs (not for books)
+         773     $0 with Record ID of corresponding Journal or Conference
+         693/710 $0 with Record ID of corresponding experiment
+    """
+    record = bfo.get_record()
+    recid = bfo.recID
+
+    # Let's add signatures
+    for field in record_get_field_instances(record, '100') + record_get_field_instances(record, '700'):
+        subfields = field_get_subfield_instances(field)
+        subfield_dict = dict(subfields)
+        if 'a' in subfield_dict:
+            author_name = subfield_dict['a']
+            rows = run_sql("SELECT personid, flag FROM aidPERSONIDPAPERS WHERE bibrec=%s AND name=%s AND flag>-2", (recid, author_name))
```

Comment:
Do we have indices in the database tables for these WHERE selectors? If not, it should be added where appropriate.